cmake/modules/RootBuildOptions.cmake

```diff
@@ -102,37 +102,37 @@ ROOT_BUILD_OPTION(cxx14 OFF "Build using C++14 compatible mode, requires gcc > 4
-ROOT_BUILD_OPTION(fitsio ON "Read images and data from FITS files, requires cfitsio")
+ROOT_BUILD_OPTION(fitsio OFF "Read images and data from FITS files, requires cfitsio")
 ROOT_BUILD_OPTION(fortran ON "Enable the Fortran components of ROOT")
 set(gcctoolchain "" CACHE PATH "Path for the gcctoolchain in case not the system gcc is used to build clang/LLVM")
 ROOT_BUILD_OPTION(gdml OFF "GDML writer and reader")
 ROOT_BUILD_OPTION(genvector ON "Build the new libGenVector library")
 ROOT_BUILD_OPTION(geocad OFF "ROOT-CAD Interface")
 ROOT_BUILD_OPTION(gfal ON "GFAL support, requires libgfal")
-ROOT_BUILD_OPTION(glite ON "gLite support, requires libglite-api-wrapper v.3 from GSI (https://subversion.gsi.de/trac/dgrid/wiki)")
+ROOT_BUILD_OPTION(glite OFF "gLite support, requires libglite-api-wrapper v.3 from GSI (https://subversion.gsi.de/trac/dgrid/wiki)")
 ROOT_BUILD_OPTION(globus OFF "Globus authentication support, requires Globus toolkit")
 ROOT_BUILD_OPTION(gnuinstall OFF "Perform installation following the GNU guidelines")
 ROOT_BUILD_OPTION(gsl_shared OFF "Enable linking against shared libraries for GSL (default no)")
-ROOT_BUILD_OPTION(gviz ON "Graphs visualization support, requires graphviz")
-ROOT_BUILD_OPTION(hdfs ON "HDFS support; requires libhdfs from HDFS >= 0.19.1")
-ROOT_BUILD_OPTION(http OFF "HTTP Server support")
+ROOT_BUILD_OPTION(gviz OFF "Graphs visualization support, requires graphviz")
+ROOT_BUILD_OPTION(hdfs OFF "HDFS support; requires libhdfs from HDFS >= 0.19.1")
+ROOT_BUILD_OPTION(http ON "HTTP Server support")
 ROOT_BUILD_OPTION(imt ON "Implicit multi-threading support")
 ROOT_BUILD_OPTION(jemalloc OFF "Using the jemalloc allocator")
-ROOT_BUILD_OPTION(krb5 ON "Kerberos5 support, requires Kerberos libs")
-ROOT_BUILD_OPTION(ldap ON "LDAP support, requires (Open)LDAP libs")
+ROOT_BUILD_OPTION(krb5 OFF "Kerberos5 support, requires Kerberos libs")
+ROOT_BUILD_OPTION(ldap OFF "LDAP support, requires (Open)LDAP libs")
 ROOT_BUILD_OPTION(libcxx OFF "Build using libc++, requires cxx11 option (MacOS X only, for the time being)")
 ROOT_BUILD_OPTION(macos_native OFF "Disable looking for libraries, includes and binaries in locations other than a native installation (MacOS only)")
 ROOT_BUILD_OPTION(mathmore ON "Build the new libMathMore extended math library, requires GSL (vers. >= 1.8)")
 ROOT_BUILD_OPTION(memory_termination OFF "Free internal ROOT memory before process termination (experimental, used for leak checking)")
-ROOT_BUILD_OPTION(memstat ON "A memory statistics utility, helps to detect memory leaks")
+ROOT_BUILD_OPTION(memstat OFF "A memory statistics utility, helps to detect memory leaks")
 ROOT_BUILD_OPTION(minuit2 OFF "Build the new libMinuit2 minimizer library")
-ROOT_BUILD_OPTION(monalisa ON "Monalisa monitoring support, requires libapmoncpp")
+ROOT_BUILD_OPTION(monalisa OFF "Monalisa monitoring support, requires libapmoncpp")
 ROOT_BUILD_OPTION(mysql ON "MySQL support, requires libmysqlclient")
```

Comment:
MySQL could also be `OFF` by default.

Reply:
Fair enough. It depends if we want to be aggressive. I would not mind removing this. I wonder if the reviewers have an opinion?

Reply:
Why not have support for Database access (if there component are findabe)?

Reply:
Users compiling this on their machines probably don't need these options (and usually have to disable by hand). It makes more sense to enable them when needed, no?

Reply:
> and usually have to disable by hand

Why do they *have* to disable them by hand?

Reply:
Well, just because mysql exists in my system doesn't mean I want ROOT with mysql enabled... The more options enabled by default, the longer it takes for users to compile. If they are not using these things, it's just a waste of time and energy.

Reply:
This is true.  On the other hand, saving that time/energy can lead to frustration in other cases.
Think of the scenario of the update-rarely user:
* download ROOT source
* struggle to read/guess the install instruction.
* build
* install
* try and be happy.
* delete build directory and source because he needs the space or what not.
* **several months** pass by
* Decide to try something out with MySQL (that she remembers installing before ROOT)
* Frustration sets in because it is surprisingly not working  and she now faced the uncertain prospect of having to restart from scratch the installation ...

Reply:
Let's not use unrealistic imagined scenarios as argument to prevent the evolution of ROOT. Searching the ROOT forum I can only find 3 posts since Aug/2015 that mention using mysql explicitly. Actually, one [post](https://root-forum.cern.ch/t/building-time-v-6-06-02-in-linux/20893) I found is for someone complaining about build time. Many more people ask questions about RooFit, and it has spinoff projects like GooFit, yet it is still OFF by default. The rationale here should be that if only a small percentage of users are using something, it should be OFF by default, and I feel that this is the case of mysql support. Also, I believe RooFit should be ON by default. The regular user is much more likely to use that than mysql support.